iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -36,7 +40,40 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
 
         private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, params DelegatingHandler[] handlers)
         {
-            _client = new IotHubGatewayServiceAPIs(uri, credentials, handlers);
+            // ConnectionLeaseTimeoutHandler is IDisposable, so save it as a field so it can be disposed later.
+            _connectionLeaseTimeoutHandler = new ConnectionLeaseTimeoutHandler(HttpTransportSettings.DefaultConnectionLeaseTimeoutMilliseconds);
+
+            // Assign each custom delegating handler as the inner handler of the previous delegating handler
+            // For instance:
+            // _connectionLeaseTimeoutHandler.InnerHandler = handlers[0],
+            // handlers[0].InnerHandler = handlers[1],
+            // handlers[1].InnerHandler = handlers[2],
+            // ...
+            // handlers[handlers.Length].InnerHandler = _httpClientHandler
+            DelegatingHandler chainedDelegatingHandler = _connectionLeaseTimeoutHandler;
+            foreach (DelegatingHandler delegatingHandler in handlers)
```

Comment:
this for-loop is building the handler pipeline from the bottom up, shouldn't we be adding the HttpClientHandler first?

Reply:
I'll be removing this chaining and passing down the delegating handler list + the one new connection lease timeout handler directly to the generated code, so this won't be an issue anymore